### PR TITLE
#12978 respect KG UOM precision when calculating package weight

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/12978

### CODE REVIEW

File: backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java [hunk 1]
```diff
@@ -180,4 +183,14 @@ public boolean hasServiceSupport(@NonNull final String shipperGatewayId)
 		return shipperRegistry.hasServiceSupport(shipperGatewayId);
 	}
 
+	private static UOMPrecision getKgUOMPrecision()
+	{
+		if (kgPrecision == null)
+		{
+			final IUOMDAO uomDAO = Services.get(IUOMDAO.class);
```
Comment (metas-ts) [inline backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java:190]: pls make this a field

File: backend/de.metas.shipper.gateway.spi/src/main/java/de/metas/shipper/gateway/spi/model/DeliveryPosition.java [hunk 2]
```diff
@@ -54,14 +56,14 @@
 	private DeliveryPosition(
 			final int repoId,
 			final int numberOfPackages,
-			final int grossWeightKg,
+			final BigDecimal grossWeightKg,
```
Comment (metas-ts) [inline backend/de.metas.shipper.gateway.spi/src/main/java/de/metas/shipper/gateway/spi/model/DeliveryPosition.java:59]: annotate it as `@NonNull`?

File: backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java [hunk 3]
```diff
@@ -120,14 +124,13 @@ private static DeliveryOrderKey createDeliveryOrderKey(
 	/**
 	 * In case the weight is <= 0, return the default value.
 	 */
-	private static int computeGrossWeightInKg(@NonNull final Collection<I_M_Package> mpackages, @SuppressWarnings("SameParameterValue") final int defaultValue)
+	private static BigDecimal computeGrossWeightInKg(@NonNull final Collection<I_M_Package> mpackages, @SuppressWarnings("SameParameterValue") final BigDecimal defaultValue)
 	{
-		final int weightInKg = mpackages.stream()
+		final BigDecimal weightInKgRaw = mpackages.stream()
```
Comment (metas-ts) [inline backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java:129]:
```suggestion
		// we don't yet have a weight-UOM in M_Package, that's why we just add up the values
		final BigDecimal weightInKgRaw = mpackages.stream()
```

**Review by metas-ts — CHANGES_REQUESTED** [review]

File: backend/de.metas.shipper.gateway.spi/src/main/java/de/metas/shipper/gateway/spi/model/DeliveryPosition.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.shipper.gateway.spi/src/main/java/de/metas/shipper/gateway/spi/model/DeliveryPosition.java]: makes sense, done.

File: backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.shipper.gateway.commons/src/main/java/de/metas/shipper/gateway/commons/ShipperGatewayFacade.java]: yup, was put off by `de.metas.shipper.gateway.commons.ShipperGatewayFacade#computeGrossWeightInKg` being static, but there's really not reason for it to be static.

**Review by metas-ts — APPROVED** [review]

**Pull request merged**